File: app/src/main/java/com/apache/fastandroid/demo/network/api/BaseResponse.java
package com.apache.fastandroid.demo.network.api;

import java.io.Serializable;

import androidx.annotation.Keep;

/**
 * Created by Jerry on 2020/11/25.
 */
@Keep
public class BaseResponse implements Serializable {
    public int code;
    public String msg;

    /**
     * 判断数据是否是有效的
     * @return
     */
    public boolean isSuccess() {
        return code == 0 && "Success".equalsIgnoreCase(msg);
    }
}

File: libnetwork/src/main/java/com/apache/fastandroid/network/response/BaseResponse.kt
package com.apache.fastandroid.network.response

import com.google.gson.annotations.SerializedName
import java.io.Serializable

open class BaseResponse<T>(var data: T  ,var errorCode: Int = -1,var code: Int = -1,var errorMsg: String = ""){
    fun isSuccess():Boolean = errorCode == 0

    fun isFailed() = !isSuccess()
    override fun toString(): String {
        return "BaseResponse(data=$data, errorCode=$errorCode, code=$code, errorMsg='$errorMsg')"
    }


}